provisioners/README.md

```diff
@@ -1,204 +1,264 @@
+		// when using git repo S2I process creates some custom maps and build pods
+		openShift.getConfigMaps()
+				.stream()
+				.filter(cfMap -> cfMap.getMetadata().getName().startsWith(spyApp.getName()))
+				.forEach(openShift::deleteConfigMap);
+		openShift.getPods()
+				.stream()
+				.filter(pod -> pod.getMetadata().getName().startsWith(spyApp.getName()))
+				.forEach(openShift::deletePod);
+	}
+    
+	@Override
+	public void scale(int replicas, boolean wait) {
+		openShift.scale(spyApp.getName(), replicas);
+		if (wait) {
+			waitForReplicas(replicas);
+		}
+	}
+
+	public void waitForReplicas(int replicas) {
+		OpenShiftWaiters.get(openShift, ffCheck).areExactlyNPodsReady(replicas, spyApp.getName()).level(Level.DEBUG)
+				.waitFor();
+		WaitersUtil.serviceEndpointsAreReady(openShift, getApplication().getName(), replicas, 8080)
+				.level(Level.DEBUG)
+				.waitFor();
+		if (replicas > 0) {
+			WaitersUtil.routeIsUp(getUrl(spyApp.getName(), false))
+					.level(Level.DEBUG)
+					.waitFor();
+		}
+	}
+
+	@Override
+	public List<Pod> getPods() {
+		return openShift.getPods(getApplication().getName());
+	}
+
+	private void deployImage() {
+            // This is an long task of many steps and thus will not be
+            // listed here.  Look at existing provisioners for examples.
+        }
 }
 ```
 
-These classes can implement their own provisioning process by using the above-mentioned annotations in order to 
-accomplish a declarative configuration, instead that requiring a specific provisioner that would instead leverage 
-a client API.
+### Step 3 Creating a new ProvisionerFactory class
+```java
+package org.secret.agents;
+
+import org.jboss.intersmash.application.Application;
+import org.jboss.intersmash.provision.ProvisionerFactory;
 
-Application instances implementing the `AutoProvisioningOpenShiftApplication` interface can be used the same ways as the 
-other `Application` types, i.e. in a testing scenario defined by a test class and its `@Intersmash` annotations.
+public class SpyProvisionerFactory implements ProvisionerFactory<SpyProvisioner> {
 
-For an example of automatic provisioning process, just run the following commands: 
-```text
-$ mvn clean package -DskipTests
-...
-$ mvn test -pl core -Dtest=AutoProvisioningTests
+	@Override
+	public SpyProvisioner getProvisioner(Application application) {
+		if (SpyApplication.class.isAssignableFrom(application.getClass()))
+			return new SpyProvisioner((SpyApplication) application);
+		return null;
+	}
+}
 ```
+
+### Step 4 Register the ProvisionerFactory as a service
+Edit service file, _./provisioners/src/main/resources/META-INF/services/org.jboss.intersmash.provision.ProvisionerFactory_ and add _org.secret.agents.SpyProvisionerFactory_ and rebuilt Intersmash.
+
+
+### Step 5 Provide a demonstration application
+The last task is to write and add a simple application that demonstrates the service working in directory, examples.
```

Comment:
I'd s/directory, examples/the `examples` directory/

Reply:
done